dpp/src/identity/state_transition/identity_credit_withdrawal_transition/apply_identity_credit_withdrawal_transition_factory.rs

```diff
@@ -0,0 +1,40 @@
+use anyhow::{anyhow, Result};
+
+use crate::{prelude::Identity, state_repository::StateRepositoryLike};
+
+use super::IdentityCreditWithdrawalTransition;
+
+pub struct ApplyIdentityCreditWithdrawalTransition<SR>
+where
+    SR: StateRepositoryLike,
+{
+    state_repository: SR,
+}
+
+impl<SR> ApplyIdentityCreditWithdrawalTransition<SR>
+where
+    SR: StateRepositoryLike,
```

Comment:
I probably would change that to an Arc, similar to other identities validators, so it's easier to share the same state repository between different validators